fili-core/src/test/groovy/com/yahoo/bard/webservice/web/endpoints/JobsServletReactiveChainforResultsEndpointSpec.groovy

```diff
@@ -152,73 +181,119 @@ class JobsServletReactiveChainforResultsEndpointSpec extends Specification {
-        mockPreResponseStore.save("ticket4", PreResponseTestingUtils.buildPreResponse("2016-04-24")).subscribe()
+        and:
+        TestSubscriber resultsSubscriber = new TestSubscriber()
 
-        and: "We start the async chain"
-        mockJobServlet.getPreResponseObservable("ticket4", apiRequest1)
-        //The delay is to ensure that we get the notification after async timeout
-        Thread.sleep(1000)
+        and:
+        jobsServlet.getResults("ticket1", apiRequest.asyncAfter).subscribe(resultsSubscriber)
 
-        and: "We receive the notification after async timeout"
-        broadcastChannel.publish("ticket4")
+        and: "We wait for the first attempt to get results from the store to come up empty before we add fake results"
+        getTicket1Latch.await(30, TimeUnit.SECONDS)
+        PreResponse results = Mock(PreResponse)
+        preResponseStore.save("ticket1", results)
 
-        then: "then we go to the PreResponseStore exactly once to get the ticket"
-        1 * mockPreResponseStore.get(_) >> Observable.just(Mock(PreResponse))
+        and: "We wait for the results to be successfully stored before sending a ready notification"
+        saveTicket1Latch.await(30, TimeUnit.SECONDS)
+
+        when: "We send a bunch of notifications through the broadcast channel, including multiples of ticket1"
+        ["ticket2", "ticket1", "ticket3", "ticket1", "ticket4", "ticket1"].each { broadcastChannel.publish(it) }
+
+        then: "The subscriber only gets the results for ticket1"
+        resultsSubscriber.assertValue(results)
+
+        cleanup:
+        preResponseStore.clearGetLatches()
+        preResponseStore.clearSaveLatches()
     }
 
-    def "If the PreResponse is available in the PreResponseStore and the notification from broadcastChannel is received within the async timeout, we go to the PreResponsestore twice"() {
+    def "If the notification from broadcastChannel is not received within async timeout, we go to the PreResponseStore exactly once"() {
         setup:
-        JobsApiRequest apiRequest1 = new JobsApiRequest(null, null, "", "", null, uriInfo, jobPayloadBuilder, apiJobStore, mockPreResponseStore, broadcastChannel)
-
-        when: "PreResponse is available in the PreResponsestore"
-        mockPreResponseStore.save("ticket4", PreResponseTestingUtils.buildPreResponse("2016-04-24")).subscribe()
+        JobsApiRequest apiRequest1 = new JobsApiRequest(
+                null,
+                "2",
+                "",
+                "",
+                null,
+                uriInfo,
+                jobPayloadBuilder,
+                apiJobStore
+        )
 
-        and: "We start the async chain"
-        mockJobServlet.getPreResponseObservable("ticket4", apiRequest1)
+        when: "We start the async chain"
+        mockJobServlet.getResults("ticket4", apiRequest1.asyncAfter)
+        //The delay is to ensure that we get the notification after async timeout
+        Thread.sleep(1000)
```

Comment:
This test is testing that things behave appropriately in the case where we have a numeric `asyncAfter`. Unfortunately, we have currently have no way of injecting latches or anything into the workflow. So our choices are to sleep (which the test was already doing before I simplified it), or refactor the JobsServlet so that we have some way of injecting additional components into the workflow.

While the second may or may not be a valid approach (depending on how much control we want to give clients over the JobServlet), it's outside the scope of this PR.